File: sdk-android-library/src/main/java/com/xyfindables/sdk/action/XYDeviceActionBuzzModern.java
package com.xyfindables.sdk.action;

import android.bluetooth.BluetoothGatt;
import android.bluetooth.BluetoothGattCharacteristic;
import android.util.Log;

import com.xyfindables.sdk.XYDevice;
import com.xyfindables.sdk.XYDeviceCharacteristic;
import com.xyfindables.sdk.XYDeviceService;

import java.util.UUID;

/**
 * Created by alex.mcelroy on 9/5/2017.
 */

public abstract class XYDeviceActionBuzzModern extends XYDeviceAction {

    private static final String TAG = XYDeviceActionBuzzModern.class.getSimpleName();

    private byte[] value;

    protected XYDeviceActionBuzzModern(XYDevice device, byte[] value) {
        super(device);
        this.value = value;
        Log.v(TAG, TAG);
    }

    @Override
    public UUID getServiceId() {
        return XYDeviceService.XY4Primary;
    }

    @Override
    public UUID getCharacteristicId() {
        return XYDeviceCharacteristic.XY4PrimaryBuzzer;
    }

    @Override
    public boolean statusChanged(int status, BluetoothGatt gatt, BluetoothGattCharacteristic characteristic, boolean success) {
        Log.v(TAG, "statusChanged:" + status + ":" + success);
        boolean result = super.statusChanged(status, gatt, characteristic, success);
        switch (status) {
            case STATUS_CHARACTERISTIC_FOUND:
                characteristic.setValue(value);
                if (!gatt.writeCharacteristic(characteristic)) {
                    result = false;
                }
                break;
        }
        return result;
    }
}

File: sdk-android-library/src/main/java/com/xyfindables/sdk/action/XYDeviceActionBuzzModern.kt
package com.xyfindables.sdk.action

import android.bluetooth.BluetoothGatt
import android.bluetooth.BluetoothGattCharacteristic
import com.xyfindables.sdk.XYDevice
import com.xyfindables.sdk.XYDeviceCharacteristic
import com.xyfindables.sdk.XYDeviceService

import java.util.UUID

/**
 * Created by alex.mcelroy on 9/5/2017.
 */

abstract class XYDeviceActionBuzzModern protected constructor(device: XYDevice, private val value: ByteArray) : XYDeviceAction(device) {

    override val serviceId: UUID
        get() = XYDeviceService.XY4Primary

    override val characteristicId: UUID
        get() = XYDeviceCharacteristic.XY4PrimaryBuzzer

    init {
        logAction(TAG, TAG)
    }

    override fun statusChanged(status: Int, gatt: BluetoothGatt?, characteristic: BluetoothGattCharacteristic?, success: Boolean): Boolean {
        logExtreme(TAG, "statusChanged:$status:$success")
        var result = super.statusChanged(status, gatt, characteristic, success)
        if (gatt !== null) {
            when (status) {
                XYDeviceAction.STATUS_CHARACTERISTIC_FOUND -> {
                    characteristic?.value = value
                    if (!gatt!!.writeCharacteristic(characteristic)) {
                        result = true
                    }
                }
            }
        }
        return result
    }

    companion object {

        private val TAG = XYDeviceActionBuzzModern::class.java.simpleName
    }
}